cpp/src/gandiva/expression_registry.h

```diff
@@ -36,22 +36,26 @@ class FunctionRegistry;
 /// data types and functions supported by Gandiva.
 class GANDIVA_EXPORT ExpressionRegistry {
  public:
-  using iterator = const NativeFunction*;
+  using nf_iterator = const NativeFunction*;
```

Comment:
better names please..its hard to follow otherwise..

Reply:
renamed nf_iterator to native_func_iterator_type and fs_iterator to func_sig_iterator_type.
thanks for reviewing